### PR TITLE
Add support for HKDF (RFC 5869).

### PR DESCRIPTION
**Pull request by Yawning**
Implementation based off the pbkdf2 code, except this is fast, and not
for password hashing.

### CODE REVIEW

**Comment by DaGenix** [comment]
Thanks for submitting this. I don't have time to take a look right now, but I'll try to take a look tomorrow.

Thanks!

File: src/hkdf.rs [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    }
+    let nbuf = [n];
+    mac.input(info);
+    mac.input(&nbuf);
+    mac.raw_result(scratch);
+    mac.reset();
+    copy_memory(output, scratch); // Sigh. :(
+}
+
+/**
```
Comment (calvinmetcalf) [inline src/hkdf.rs]: doc comments in rust use ///'s not /**  see [this](http://doc.rust-lang.org/book/documentation.html)

**Comment by Yawning** [comment]
@calvinmetcalf Fixed.  Still getting a hang of this rust thing.

I also went and changed the HKDF-Expand code to eliminate the helper function, since unlike PBKDF2, just generating `T(N)` in the loop body is compact enough.  The newer code is easier to read/maintain I hope.

**Comment by DaGenix** [comment]
Thanks!

**Pull request merged**